ACE/ace/OS_NS_arpa_inet.cpp

```diff
@@ -99,4 +102,24 @@ ACE_OS::inet_aton (const char *host_name, struct in_addr *addr)
 #endif  /* ACE_LACKS_INET_ATON */
 }
 
+#if defined (ACE_INTEGRITY) && defined (ACE_LACKS_INET_NTOA)
+char* ACE_OS::inet_ntoa (const struct in_addr addr)
+{
+  ACE_OS_TRACE ("ACE_OS::inet_ntoa");
+
+  static ACE_thread_mutex_t mutex;
+  static char addrstr[INET_ADDRSTRLEN + 1] = { 0 };
```

Comment:
The lock makes this thread safe within this function, but you are returning a static buffer so threads could overwrite this, shouldn't the buffer be a in thread local storage?

Reply:
Or `inet_ntoa` can be non-thread-safe (https://pubs.opengroup.org/onlinepubs/9699919799.orig/) and the caller will need to be aware of that?

Reply:
There's a lot of added complexity in going to a thread-local; and the lack of safety in going with non-thread-safe (even though POSIX allows it) doesn't look good either.

How about just ACE_LACKS_INET_NTOA for this platform?  It doesn't seem to be used in core ACE+TAO libs and inet_ntop is a better alternative.

Reply:
If I remember correctly, there were places in TAO when the ORB is initialized that this function is called. But that was the older TAO version. I'm not sure if current TAO still uses it.

Reply:
I tested locally with this emulation function removed and it doesn't look like TAO is using it. So I removed it and `inet_ntoa` now just returns unsupported with ACE_LACKS_INET_NTOA on this platform. @jwillemsen